src/condition.cpp

```diff
@@ -2173,9 +2186,9 @@ conditional_t::get_set_dbl( const J &jo, const std::optional<dbl_or_var_part> &m
                 var_name = get_talk_varname( jo, "var_name", false, empty );
             }
             return [is_npc, var_name, type, min, max]( dialogue const & d, double input ) {
-                write_var_value( type, var_name, d.actor( is_npc ), std::to_string( handle_min_max( d, input,
-                                 min,
-                                 max ) ) );
+
+                write_var_value( type, var_name, d.actor( is_npc ),
+                                 remove_trailing_zeroes( std::to_string( handle_min_max( d, input, min, max ) ) ) );
```

Comment:
```suggestion
                write_var_value( type, var_name, d.actor( is_npc ),
                                 // NOLINTNEXTLINE(cata-translate-string-literal)
                                 string_format( "%g", handle_min_max( d, input, min, max ) ) );
```
Uses 6 digits of precision and strips trailing zeroes and decimal point.